backend/cmm_helpers.ml

```diff
@@ -1793,19 +1856,17 @@ let make_mixed_alloc ~mode dbg ~tag ~value_prefix_size args args_memory_chunks =
           (* regular scanned part of a block *)
           match memory_chunk with
           | Word_int | Word_val -> ok ()
-          | Byte_unsigned | Byte_signed | Sixteen_unsigned | Sixteen_signed ->
-            error "mixed blocks"
+          | Byte_unsigned | Byte_signed | Sixteen_unsigned | Sixteen_signed
```

Comment:
Same remark as another change: this changes semantics /allow more things. It may be correct/reasonable, but it's surprising to see this as part of a refactor PR.